### PR TITLE
chore(chain, cmd, dot): Change retain blocks type to be `uint32`

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Changes

Changed retain blocks type to `uint32` and 
modified the code to use `cli.Uint64Flag` struct instead of `cli.Int64Flag` struct.

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

#2839

## Primary Reviewer

@timwu20

### CODE REVIEW

File: cmd/gossamer/config.go [hunk 1]
```diff
@@ -485,7 +485,7 @@ func setDotGlobalConfigFromFlags(ctx *cli.Context, cfg *dot.GlobalConfig) error
 		cfg.MetricsAddress = metricsAddress
 	}
 
-	cfg.RetainBlocks = ctx.Int64(RetainBlockNumberFlag.Name)
+	cfg.RetainBlocks = uint32(ctx.Uint64(RetainBlockNumberFlag.Name))
```
[the remark this answers is not in the document]
Comment (qdm12) [reply cmd/gossamer/config.go:488]: Probably error out I'd say in this case, that warn log might get unnoticed with unwanted consequences.

File: cmd/gossamer/main.go [hunk 2]
```diff
@@ -425,7 +425,7 @@ func pruneState(ctx *cli.Context) error {
 	}
 
 	bloomSize := ctx.GlobalUint64(BloomFilterSizeFlag.Name)
-	retainBlocks := ctx.GlobalInt64(RetainBlockNumberFlag.Name)
+	retainBlocks := uint32(ctx.GlobalUint64(RetainBlockNumberFlag.Name))
```
Comment (qdm12) [inline cmd/gossamer/main.go:428]: We should add a check that the value `ctx.GlobalUint64(RetainBlockNumberFlag.Name)` is less than the max uint32 (`^uint32(0)`)

**Review by qdm12 — APPROVED** [review]
Nice, just add these 2 checks in case the value is above max uint32 and let's get this merged!

Pre-approving :wink:

File: cmd/gossamer/config.go [hunk 3]
```diff
@@ -485,7 +485,14 @@ func setDotGlobalConfigFromFlags(ctx *cli.Context, cfg *dot.GlobalConfig) error
 		cfg.MetricsAddress = metricsAddress
 	}
 
-	cfg.RetainBlocks = ctx.Int64(RetainBlockNumberFlag.Name)
+	const uint32Max = ^uint32(0)
+	flagValue := ctx.Uint64(RetainBlockNumberFlag.Name)
+
+	if uint64(uint32Max) < flagValue {
+		return fmt.Errorf("retain blocks value overflows uint32 boundaries, must be less than, %d", uint32Max)
```
Comment (EclesioMeloJunior) [inline cmd/gossamer/config.go:492]: supernit
```suggestion
		return fmt.Errorf("retain blocks value overflows uint32 boundaries, must be less or equal than: %d", uint32Max)
```

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

**Review by EclesioMeloJunior — APPROVED** [review]

**Pull request merged**